mava/systems/jax/launcher.py

```diff
@@ -159,24 +170,38 @@ def launch(self) -> None:
 
             lp.launch(
                 self._program,
-                lp.LaunchType.LOCAL_MULTI_PROCESSING,
+                launch_type=self._lp_launch_type,
                 terminal=self._terminal,
                 local_resources=local_resources,
             )
         else:
             episode = 1
             executor_steps = 0
 
-            _ = self._node_dict["data_server"]
+            data_server = self._node_dict["data_server"]
             _ = self._node_dict["parameter_server"]
             executor = self._node_dict["executor"]
             evaluator = self._node_dict["evaluator"]
             trainer = self._node_dict["trainer"]
 
-            while True:
-                executor_stats = executor.run_episode_and_log()
+            # getting the maximum queue size
+            queue_threshold = data_server.server_info()["trainer"].max_size
 
-                if episode % self._sp_trainer_period == 0:
+            while True:
+                # if the queue is too full we skip the executor to ensure that the
+                # executor won't hang when trying to push experience
+                if data_server.server_info()["trainer"].current_size < int(
+                    queue_threshold * 0.75
+                ):
```

Comment:
Do we need this check? PPO uses a queue, so each element get's removed as it is used. The max `current_size` I could get when running the example was ~ 7, when queue's `.max_size` is 500. 

Basically, was there an issue this is solving or just a precaution? I think the reverb params like `max_times_sampled` should be handling these sort of things.

What do you think?

Reply:
This is the `launcher` for all systems, so will this not be useful when we add other systems?

Reply:
This prevents Reverb from pausing the executor when the queue fills up. In this case the queue does not fill up yes. But it does fill up under other settings. The same thing with the queue is empty when the trainer samples. We can't let Reverb handle it internally as Reverb just waits which hangs the entire training process.

Reply:
Does a `trainer.step` always reduce the size of the table? Even when are not using a queue?

Reply:
`trainer.step `hangs if there is less experience in the `data_server` than is requested by the trainer sample. This is not just for the queue, but also for other table types. So if the trainer samples more data than gets generated at some point it will hang. Does that make sense?